packet.go

```diff
@@ -100,21 +122,32 @@ func (p *Packet) decodeArp() {
 	arp.HwAddressSize = pkt[4]
 	arp.ProtAddressSize = pkt[5]
 	arp.Operation = binary.BigEndian.Uint16(pkt[6:8])
-	arp.SourceHwAddress = pkt[8 : 8+arp.HwAddressSize]
-	arp.SourceProtAddress = pkt[8+arp.HwAddressSize : 8+arp.HwAddressSize+arp.ProtAddressSize]
-	arp.DestHwAddress = pkt[8+arp.HwAddressSize+arp.ProtAddressSize : 8+2*arp.HwAddressSize+arp.ProtAddressSize]
-	arp.DestProtAddress = pkt[8+2*arp.HwAddressSize+arp.ProtAddressSize : 8+2*arp.HwAddressSize+2*arp.ProtAddressSize]
+	if arp.HwAddressSize != 0 && len(pkt) >= int(8+2*arp.HwAddressSize+2*arp.ProtAddressSize) {
+		arp.SourceHwAddress = pkt[8 : 8+arp.HwAddressSize]
+		arp.SourceProtAddress = pkt[8+arp.HwAddressSize : 8+arp.HwAddressSize+arp.ProtAddressSize]
+		arp.DestHwAddress = pkt[8+arp.HwAddressSize+arp.ProtAddressSize : 8+2*arp.HwAddressSize+arp.ProtAddressSize]
+		arp.DestProtAddress = pkt[8+2*arp.HwAddressSize+arp.ProtAddressSize : 8+2*arp.HwAddressSize+2*arp.ProtAddressSize]
+
+		p.setHeader(arp)
 
-	p.Headers = append(p.Headers, arp)
-	p.Payload = p.Payload[8+2*arp.HwAddressSize+2*arp.ProtAddressSize:]
+		p.Payload = p.Payload[8+2*arp.HwAddressSize+2*arp.ProtAddressSize:]
+	} else {
+		dlog.Infof("s: %d L:%d(H:%d P:%d) -> T:%x D:%x S:%x A:%x P:%x %x", p.Seq, len(p.Payload), arp.HwAddressSize, arp.ProtAddressSize, p.Type, p.DestMac, p.SrcMac, arp.Addrtype, arp.Protocol, bytes.Split(p.Data, []byte(",")))
```

Comment:
I don't think logging will add much here, can't return an error due to the function's signature... so silently ignore?